### PR TITLE
Minor fixes and improvements to docs

### PR DESCRIPTION
**Pull request by denniskigen**
- Fix a broken link to `Prerequisites` in the `Setup` guide.
- Add [The Modern JavaScript Tutorial](https://javascript.info) as a suggested JavaScript resource.
- Change bolded content to a relative link in the `What is Carbon?` section of the `Using Carbon and the Styleguide` guide.
- Update Zeplin description in the `How to Use Zeplin` section of the `Using Carbon and the Styleguide` guide.
- Add a relative link to the `Contributing` guide in the `Support in other frameworks` section of the `Config` guide.
- Add a `catch` block to the `useEffect hook` example in the `Retrieving and posting data` guide.
- Add a bit about automatic extraction of translations using `i18next-parser` in the `How translation works` guide.
- Small grammatical fixes.

### CODE REVIEW

File: docs/main/creating_a_microfrontend.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ Once you've created a microfrontend, you'll want to integrate it into your
 application. There are two steps for doing so.
 
 First, publish your microfrontend package to NPM. See
-[Release and Deployment](../getting_started/release_and_deployment)
+[Release and Deployment](getting_started/release_and_deployment)
```
Comment (FlorianRappl) [inline docs/main/creating_a_microfrontend.md:14]: link seems to be wrong. The previous one was (almost; .md missing) right.

File: docs/main/carbon.md [hunk 2]
```diff
@@ -7,27 +7,26 @@ interface guidelines, and a community of contributors.
 
 OpenMRS 3.0 uses Carbon to create a unified UI. Third party Design Systems like Carbon 
 are kept professionally maintained by another community which reduces work for OpenMRS 
-contributors. (More details below in the section *Why use a Style Guide or a Design System?*)
+contributors. (More details in the [Why Use a Style Guide or a Design System](/main/carbon?id=why-use-a-style-guide-or-a-design-system) section below.
```
Comment (FlorianRappl) [inline docs/main/carbon.md:10]: link is wrong -> use `#why-use-...` as anchor-only link or `./carbon#why-use-...`.

File: docs/main/config.md [hunk 3]
```diff
@@ -288,7 +288,7 @@ defaults for configuration elements for which no values have been provided.
 
 #### Support in other frameworks (Angular, Vue, Svelte, etc.)
 
-This hasn't been implemented yet, but we would like to implement it! See "Contributing"
+This hasn't been implemented yet, but we would like to implement it! See [Contributing](/getting_started/contributing).
```
Comment (FlorianRappl) [inline docs/main/config.md:291]: Link is wrong. Should be `../getting_started/contributing.md`.

**Review by FlorianRappl — CHANGES_REQUESTED** [review]
LGTM - just the links ...

Keep in mind that links also *need* to be working on GitHub / when the markdown is rendered directly. That is, actually, the primary concern. The rendering in some tool is secondary, and here - if the link is done wrongly - the tool must be adjusted. Not our links.

File: docs/getting_started/setup.md [hunk 4]
```diff
@@ -113,7 +113,7 @@ npm run serve --https
 
 The protocol of the application must match the protocol of the locally-served microfrontend.
 
-This command will serve the microfrontend and tell you the port where it is serving,
+This command will serve the microfrontend and tell you the port where it is serving it at,
```
Comment (brandones) [inline docs/getting_started/setup.md:116]:
```suggestion
This command will serve the microfrontend and tell you the port where it is serving,
```

File: docs/main/config.md [hunk 5]
```diff
@@ -288,7 +288,7 @@ defaults for configuration elements for which no values have been provided.
 
 #### Support in other frameworks (Angular, Vue, Svelte, etc.)
 
-This hasn't been implemented yet, but we would like to implement it! See "Contributing"
+This hasn't been implemented yet, but we would like to implement it! See [Contributing](../getting_started/contributing).
```
Comment (brandones) [inline docs/main/config.md:291]:
```suggestion
This hasn't been implemented yet, but we would like to implement it! See [Contributing](../getting_started/contributing.md).
```

**Review by brandones — APPROVED** [review]
Looks great, thanks so much @denniskigen ! Approved modulo my two suggestions above.

**Review by FlorianRappl — APPROVED** [review]
Thanks for the changes. Fully agreed with @brandones - same approval (just commit the suggested changes and here we go!).

**Comment by brandones** [comment]
Also I added a note about how we're dealing with links to the README page. Just codifying what you guys said here. https://github.com/openmrs/openmrs-esm-core/commit/9a42f938eaa4299cd96034114eabe16bb6a5ec71

**Pull request merged**